sklearn/impute.py

```diff
@@ -1207,6 +1207,14 @@ def fit(self, X, y=None):
             raise ValueError("'features' has to be either 'missing-only' or "
                              "'all'. Got {} instead.".format(self.features))
 
+        if sparse.issparse(X):
+            # missing_values = 0 not allowed with sparse data as it would
```

Comment:
Since `fit` and `transform` both call `_validate_imput`, we can move this test into this function to avoid redundant code.

Reply:
Good idea. Thanks!